src/app/state/settings_state.rs

```diff
@@ -0,0 +1,40 @@
+use crate::app::state::{AppAction, AppEvent, UpdatableState};
+use crate::player::SpotifyPlayerSettings;
+
+#[derive(Clone, Debug)]
+pub enum SettingsAction {
+    ChangePlayerSettings(SpotifyPlayerSettings),
+}
+
+impl From<SettingsAction> for AppAction {
+    fn from(settings_action: SettingsAction) -> Self {
+        Self::SettingsAction(settings_action)
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum SettingsEvent {
+    PlayerSettingsChanged(SpotifyPlayerSettings),
+}
+
+impl From<SettingsEvent> for AppEvent {
+    fn from(settings_event: SettingsEvent) -> Self {
+        Self::SettingsEvent(settings_event)
+    }
+}
+
+#[derive(Default)]
+pub struct SettingsState {}
```

Comment:
I think  you could store the Settings here as mentioned at some point :)

That way, the action could have no payload with it (just `ChangePlayerSettings`) and whenever that action is dispatched, we just do `self.settings = SpotSettings::new_from_gsettings()` since it's already synced probably? 

that way you don't have to do the mapping from the Adwaita widgets to SpotSettings everytime

Reply:
Makes sense. Moved SpotSettings into SettingsState and SettingsWindow use it whenever possible, while there are still some redundant `SpotSettings::new_from_gsettings()` calls from the player and `update_with` handlers for SettingsState.

Reply:
I am nitpicking a bit, but I think one action is enough here, and the state should decide whether to return no event or `vec![SettingsEvent::PlayerSettingsChanged]` if the player part of the settings have changed

Reply:
`set_player_settings` still has to emit `PlaybackAction::Stop` to stop playing (otherwise there will be some inconsistency in UI), so the two same `if`s are put in `update_with` and `Settings::new`. It's not a nitpick because I had the same concern, just thought of avoiding the duplicate ifs at first.

Reply:
Oh you're right absolutely!